src/components/Header/NavComponents/userprofile.tsx

```diff
@@ -1,15 +1,53 @@
-import React, { useMemo } from "react";
+import React, { useMemo, useState } from "react";
 import {
   Text,
   Center,
   Badge,
   useColorModeValue as mode,
+  Button,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverBody,
+  Stack,
+  PopoverArrow,
+  Heading,
 } from "@chakra-ui/react";
 import { bigNumberToString } from "../../../utils/fixedPoint";
 import { useAmountAvailableToStake } from "../../../queries/amountAvailableToStake";
 import { useAppWeb3 } from "../../../hooks/appWeb3";
+import { useChainAddresses } from "../../../utils/chainAddresses";
+import { changeChain } from "../../../utils/changeChain";
+import { internalAddressesPerNetworkId } from "../../../utils/contracts/contractAddresses/internalAddresses";
+import { UnsupportedChainIdError, useWeb3React } from "@web3-react/core";
+
+import {
+  frameConnector,
+  injectedConnector,
+  walletConnectConnector,
+} from "../../../hooks/injectedConnectors";
 
 export const UserProfile: React.FC<{}> = () => {
+  const [connector, setConnector] = useState("");
+
+  injectedConnector
+    .getAccount()
+    .then((res: any) => {
+      setConnector("metamask");
+    })
+    .catch((err: any) => {});
+  walletConnectConnector
+    .getAccount()
+    .then((res: any) => {
+      setConnector("walletconnect");
+    })
+    .catch((err: any) => {});
+  frameConnector
+    .getAccount()
+    .then((res: any) => {
+      setConnector("frame");
+    })
+    .catch((err: any) => {});
```

Comment:
What are you even trying to do here? `const { account, library } = useAppWeb3();` and you'll have context. This is an effectful sequence of async calls being performed on every render.

Reply:
I tried using `library` but it sometimes crashed when I tried accessing to it's values, but I think I found a way to prevent this

Reply:
`library` is undefined if there's a connector in an unavailable chain, I used this instead:
```
const { connector } = useWeb3React();
const isMetamask = connector instanceof InjectedConnector;
```

Reply:
That's an acceptable way to do it- you can also cast it to an imported type [`AbstractConnector`](https://www.npmjs.com/package/@web3-react/abstract-connector) which will let you use it regardless of which connector type it is.

The crashing when you tried accessing its values was because the fields you were accessing were not defined on the types involved; [User-defined type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) are the "right" way to do that if simple definedness and abstract-interface usage are insufficient. Your current method is plenty viable and acceptable though.